### ISSUE TITLE
Crash after refreshing page when logged in via broker.demo-ess.inrupt.com

### ISSUE THREAD

**Issue by RubenVerborgh** [inrupt/solid-client-authn-js#685]
### Search terms you've used

invalid token endpoint

### To Reproduce
1. Execute this code: https://docs.inrupt.com/developer-tools/javascript/client-libraries/tutorial/authenticate/#example
2. Log in (I did so with a Google account)
3. You are now redirected to the page
4. Refresh
5. `handleIncomingRedirect` will crash and log in the console
6. Remains on subsequent refreshes; have to clear local storage

You can reproduce it, for instance, with the `packages/browser/examples/single/bundle/` app (on tag v1.1.0).

### Expected result

Either I am logged in (preferred) or logged out (also acceptable), but no crash/

### Actual result

```
Uncaught (in promise) Error: Invalid token endpoint response (missing the field 'access_token'): {"error":"invalid_grant","error_description":"JpaAuthorizationCodeRepository: no authorization code found for value 3WNFMCmqHVhGIuXEpJfxmp"}
    validateTokenEndpointResponse webpack:///./node_modules/@inrupt/oidc-client-ext/dist/index.es.js?:206
    getTokens webpack:///./node_modules/@inrupt/oidc-client-ext/dist/index.es.js?:244
    getDpopToken webpack:///./node_modules/@inrupt/oidc-client-ext/dist/index.es.js?:276
    exchangeDpopToken webpack:///./node_modules/@inrupt/solid-client-authn-browser/dist/login/oidc/redirectHandler/AuthCodeRedirectHandler.js?:22
    handle webpack:///./node_modules/@inrupt/solid-client-authn-browser/dist/login/oidc/redirectHandler/AuthCodeRedirectHandler.js?:64
    handle webpack:///./node_modules/@inrupt/solid-client-authn-core/dist/util/handlerPattern/AggregateHandler.js?:26
    handleIncomingRedirect webpack:///./node_modules/@inrupt/solid-client-authn-browser/dist/ClientAuthentication.js?:51
    handleIncomingRedirect webpack:///./node_modules/@inrupt/solid-client-authn-browser/dist/Session.js?:34
```

### Environment
Please run

  npx envinfo --system --npmPackages --binaries --npmGlobalPackages --browsers

in your project folder and paste the output here:

```
  System:
    OS: macOS Mojave 10.14.6
    CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
    Memory: 468.27 MB / 16.00 GB
    Shell: 5.0.18 - /usr/local/bin/bash
  Binaries:
    Node: 12.18.0 - ~/.nvm/versions/node/v12.18.0/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.7 - ~/.nvm/versions/node/v12.18.0/bin/npm
  Browsers:
    Chrome: 86.0.4240.198
    Firefox: 83.0
    Safari: 13.1.1
  npmPackages:
    @inrupt/solid-client-authn-browser: ^1.1.0 => 1.1.0
    webpack: ^4.44.2 => 4.44.2
    webpack-cli: ^4.2.0 => 4.2.0
```

**Comment by RubenVerborgh** [comment]
A problem seems to be that the redirect query string does not clear; so refreshing sends it again.
Manually removing it results in being logged out and no crash/

**Comment by NSeydoux** [comment]
Thanks for reporting this Ruben ! This should be now resolved, I just released v1.1.1 that contains the fix for the issue, let me know if it's still happening :)

**Comment by RubenVerborgh** [comment]
@pmcb55 @NSeydoux Thanks, conformed working.

However, you might want to remove the query string _first_ before all processing happens. Right now, there is about .5s of the query string being unnecessarily visible: https://rubenverborgh.github.io/solid-patch-tool/

**Comment by RubenVerborgh** [comment]
Actually, there's a another, more conceptual issue now.

We are calling `session.handleIncomingRedirect` with `window.location.href` as a parameter, which makes sense. Testability etc.

However, when erasing the query parameter with `history.replaceState`, we actually pretend to know that `handleIncomingRedirect` _was_ called specifically with `window.location.href`, which we are not sure.

Therefore, I would propose to:
a. allow `handleIncomingRedirect` to be called without a parameter
  - in that case, it defaults to `window.location.href` and clears history
b. allow `handleIncomingRedirect` to be called with a parameter
  - in that case, it leaves history alone

I also wonder whether, in case B, we should protect against the `Invalid token endpoint response` error above. What I would want to see is "token already used" or "token no longer valid"; this error message is cryptic.

**Comment by NSeydoux** [comment]
You are absolutely right, I'll see what error message the token endpoint returns exactly, and how to bubble it up for more clarity. Allowing handleincomingRedirect to be called without parameters is a good idea, I'll create a ticket for this.

**Comment by nicolasmondada** [comment]
This was addressed in #695
